### PR TITLE
Add initial V2 readme

### PR DESCRIPTION
**Pull request by alxkzmn**

### CODE REVIEW

**Review by sifnoc — CHANGES_REQUESTED** [review]
I think the link given under the Univariate Grand Sum Calculation section covers most of the details.

File: kzg_prover/README.md [hunk 1]
```diff
@@ -1 +1,52 @@
-# TODO
+# Summa V2: Polynomial Encoding Approach
+
+## Motivation
+
+Summa V1 was using a Merkle sum tree (MST) as the main data structure and a cryptographic commitment. MST that has $n$ leaves involves hashing operations over $2n-1$ entries, making it computationally demanding. Additionally, the MST inclusion proofs in Summa V1 have to be wrapped into a ZK-SNARK, making it computationally demanding to generate all of them at once for the entire user base of the Custodian (~100M users).
+
+## Univariate Grand Sum Calculation
+
+The grand total of all the Custodian's $n$ user cryptocurrency balances is the Custodian's liabilities $S$. Summa V2 is using a property of the _sum of all roots of unity in a finite field_ being _equal to zero_ to find the liabilities. This property allows to efficiently calculate the grand sum of univariate polynomial evaluations. Summa V2 takes advantage of that by encoding the user balances into a univariate polynomial in a special way. The resulting proof of solvency protocol has the following steps:
```
[the remark this answers is not in the document]
Comment (alxkzmn) [reply kzg_prover/README.md:9]: This is to make the font _italic_.

File: kzg_prover/README.md [hunk 2]
```diff
@@ -1 +1,52 @@
...
+
+1. construct a polynomial of degree $d = n - 1$ that interpolates the points ($\omega^i$,$b_i$) where $i \in 0..n-1$ is the user index, $\omega^i$ is the power of an $n$-th primitive root of unity ($x$ value), and $b_i$ is the $i$-th user balance value ($y$ value);
```
[the remark this answers is not in the document]
Comment (alxkzmn) [reply kzg_prover/README.md:11]: I'll add links to the code.

File: kzg_prover/README.md [hunk 3]
```diff
@@ -1 +1,52 @@
-# TODO
+# Summa V2: Polynomial Encoding Approach
+
+## Motivation
+
+Summa V1 was using a Merkle sum tree (MST) as the main data structure and a cryptographic commitment. MST that has $n$ leaves involves hashing operations over $2n-1$ entries, making it computationally demanding. Additionally, the MST inclusion proofs in Summa V1 have to be wrapped into a ZK-SNARK, making it computationally demanding to generate all of them at once for the entire user base of the Custodian (~100M users).
```
Comment (enricobottazzi) [inline kzg_prover/README.md:5]: Add link to release version of Summa v1.1 when mentioning it

Comment (enricobottazzi) [reply kzg_prover/README.md:5]: I would say "MST that has $n$ leaves involves $2n-1$ hashing operations, making it computationally demanding. Additionally, the MST inclusion proofs in Summa V1 have to be wrapped into a ZK-SNARK, making it **unfeasible** to generate all of them at once for the entire user base of the Custodian (~100M users).

File: kzg_prover/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (enricobottazzi) [reply kzg_prover/README.md:11]: $b_i$ has some formatting error

File: kzg_prover/README.md [hunk 4]
```diff
@@ -1 +1,52 @@
...
+- the advice columns in Halo2 are interpolated as polynomials, and the $x$ values are the powers of a primitive root of unity;
+- Halo2 is internally using a KZG polynomial commitment scheme;
+- the use of a ZK-SNARK allows to additionally constrain the balance values, namely, perform the range check.
+
+The algorithm works as follows:
+
+1. Assign all the user balances to an unblinded advice column of the [circuit](../kzg_prover/src/circuits/univariate_grand_sum.rs). The unblinded advice column is a kind of advice column without the random values (blinding factors) added at the bottom.
+2. Generate the ZK-SNARK proof for the circuit, effectively encoding the balance values into a polynomial and performing a KZG commitment to this polynomial.
+3. Perform a KZG opening proof of the polynomial at $x=0$ and publicly reveal the constant term $a_0$ of the polynomial. The public can then calculate the liabilities by multiplying the $a_0$ by $d + 1$ where $d$ is the polynomial degree.
+4. Privately provide to each user a KZG opening proof of the corresponding user opening.
```
Comment (enricobottazzi) [inline kzg_prover/README.md:30]: Explain what the user opening corresponds to: namely opening of the username + opening of the balances

File: kzg_prover/README.md [hunk 5]
```diff
@@ -1 +1,52 @@
...
+
+Halo2 allows to efficiently implement the described algorithm for the following reasons:
+
+- the advice columns in Halo2 are interpolated as polynomials, and the $x$ values are the powers of a primitive root of unity;
+- Halo2 is internally using a KZG polynomial commitment scheme;
+- the use of a ZK-SNARK allows to additionally constrain the balance values, namely, perform the range check.
+
+The algorithm works as follows:
+
+1. Assign all the user balances to an unblinded advice column of the [circuit](../kzg_prover/src/circuits/univariate_grand_sum.rs). The unblinded advice column is a kind of advice column without the random values (blinding factors) added at the bottom.
```
Comment (enricobottazzi) [inline kzg_prover/README.md:27]: Can you explain why this caracteristic of unblinded advice column is useful for our purpose?

**Review by enricobottazzi — CHANGES_REQUESTED** [review]

File: kzg_prover/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sifnoc) [reply kzg_prover/README.md:9]: I see, Thanks for tip!

File: kzg_prover/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sifnoc) [reply kzg_prover/README.md:11]: It looks correctly displayed on my ide, but broken in github.

**Review by enricobottazzi — APPROVED** [review]

**Review by sifnoc — APPROVED** [review]

**Pull request merged**